### PR TITLE
Add spelling checker and link checker

### PR DESCRIPTION
**Pull request by brian-avery**

### CODE REVIEW

**Comment by brian-avery** [comment]
Fixes https://github.com/Maistra/maistra.github.io/issues/107

File: prow/gen-config.sh [hunk 1]
```diff
@@ -4,6 +4,12 @@ set -ex
 
 # re-generate config
 rm config.gen.yaml
+echo "#======================================
+# This configuration is auto-generated. 
+# To update:
+#    Modify files in the config directory
+#    Run gen-config.sh to regenerate.
+#======================================" >> config.gen.yaml
```
Comment (dgn) [inline prow/gen-config.sh:12]: Thanks, this makes sense

File: docker/tests/checkLinks.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+set -e -x
+
+export LC_ALL=C.UTF-8
+hugo -D
+cd public
+htmlproofer --url-ignore "#","http://prometheus-istio-system.127.0.0.1.nip.io","http://getgrav.org" --assume-extension --check-external-hash .
```
Comment (dgn) [inline docker/tests/checkLinks.sh:7]: I think this should live in the maistra.github.io repo, made accessible by e.g. a Makefile. Reason being that things like the `url-ignore` strongly depend on content

File: docker/tests/runLinter.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+set -e -x
+vale .
```
Comment (dgn) [inline docker/tests/runLinter.sh:3]: same here, I think this can go into a `Makefile` in the maistra.github.io repo

Comment (brian-avery) [reply docker/tests/runLinter.sh]: Thanks. Fixed in https://github.com/Maistra/maistra.github.io/pull/123

File: docker/tests/checkLinks.sh (hunk 2 above)
Comment (brian-avery) [reply docker/tests/checkLinks.sh]: Thanks. Fixed in Maistra/maistra.github.io#123

File: docker/maistra-builder_1.1.Dockerfile [hunk 4]
```diff
@@ -12,6 +12,7 @@ RUN chmod -R +x /tmp/scripts/ && \
     /tmp/scripts/install_shellcheck.sh && \
     /tmp/scripts/install_kubectl.sh && \
     /tmp/scripts/install_docker.sh && \
+    /tmp/scripts/install_docs_tools.sh \
```
Comment (dgn) [inline docker/maistra-builder_1.1.Dockerfile:15]: this seems to be missing a `&&` at the end - this way we're just passing the `rm` command as an arg to the script

Comment (brian-avery) [reply docker/maistra-builder_1.1.Dockerfile:15]: Good catch :) Built properly but I hadn't checked to see if the directory exists.

Comment (brian-avery) [reply docker/maistra-builder_1.1.Dockerfile]: Fixed

**Pull request merged**